tracer/src/Datadog.Trace/AppSec/WafEncoding/Encoder.cs

```diff
@@ -66,268 +68,62 @@ public static string FormatArgs(object o)
 
         public IEncodeResult Encode<TInstance>(TInstance? o, int remainingDepth = WafConstants.MaxContainerDepth, string? key = null, bool applySafetyLimits = true)
         {
-            var lstPointers = new List<IntPtr>();
-            var pool = Pool;
-            var result = Encode(o, lstPointers, remainingDepth, key, applySafetyLimits, pool: pool);
-            return new EncodeResult(lstPointers, pool, ref result);
+            var context = new EncoderContext(remainingDepth, applySafetyLimits, Pool);
+            var result = Encode(ref context, key, o);
+            return new EncodeResult(context.Buffers, context.Pool, ref result);
         }
 
-        public unsafe DdwafObjectStruct Encode<TInstance>(TInstance? o, List<IntPtr> argToFree, int remainingDepth = WafConstants.MaxContainerDepth, string? key = null, bool applySafetyLimits = true, UnmanagedMemoryPool? pool = null)
+        // -----------------------------------
```

Comment:
what is this line for?

Reply:
It's just to separate public API from internals